v2/rest/client.go

```diff
@@ -0,0 +1,310 @@
+	"github.com/SKF/go-utility/v2/stages"
+	"github.com/SKF/go-utility/v2/uuid"
+)
+
+type TreeFilter struct {
+	Depth         int
+	Limit         int
+	Offset        int
+	MetadataKey   string
+	MetadataValue string
+	NodeTypes     []string
+	ModifiedAfter string
+}
+
+type HierarchyClient interface {
+	GetNode(ctx context.Context, id uuid.UUID) (models.GetNodeResponse, error)
+	CreateNode(ctx context.Context, node models.CreateNodeRequest) (models.CreateNodeResponse, error)
+	UpdateNode(ctx context.Context, id uuid.UUID, node models.UpdateNodeRequest) (models.UpdateNodeResponse, error)
+	DeleteNode(ctx context.Context, id uuid.UUID) error
+	DuplicateNode(ctx context.Context, source uuid.UUID, destination uuid.UUID, suffix string) (models.DuplicateNodeResponse, error)
+
+	GetAncestors(ctx context.Context, id uuid.UUID, height int, nodeTypes ...string) (models.GetAncestorsResponse, error)
+	GetCompany(ctx context.Context, id uuid.UUID) (models.GetCompanyResponse, error)
+	GetSubtree(ctx context.Context, id uuid.UUID, filter TreeFilter, continuationToken string) (models.GetSubtreeResponse, error)
+	GetSubtreeCount(ctx context.Context, id uuid.UUID, nodeTypes ...string) (models.GetSubtreeCountResponse, error)
+
+	LockNode(ctx context.Context, id uuid.UUID, recursive bool) error
+	UnlockNode(ctx context.Context, id uuid.UUID, recursive bool) error
+
+	GetOrigins(ctx context.Context, provider, continuationToken string, limit int) (models.GetOriginsResponse, error)
+	GetOriginsByType(ctx context.Context, provider, originType, continuationToken string, limit int) (models.GetOriginsResponse, error)
+	GetProviderNodeIDs(ctx context.Context, provider, continuationToken string, limit int) (models.GetNodesByPartialOriginResponse, error)
+	GetProviderNodeIDsByType(ctx context.Context, provider, originType, continuationToken string, limit int) (models.GetNodesByPartialOriginResponse, error)
+	GetOriginNodeID(ctx context.Context, origin models.Origin) (models.GetNodeByOriginResponse, error)
+}
+
+type client struct {
+	*rest.Client
+	clientID string
+}
+
+func WithStage(stage string) rest.Option {
+	if stage == stages.StageProd {
+		return rest.WithBaseURL("https://api.hierarchy.enlight.skf.com")
+	}
+
+	return rest.WithBaseURL(fmt.Sprintf("https://api.%s.hierarchy.enlight.skf.com", stage))
+}
+
+func NewClient(clientID string, opts ...rest.Option) HierarchyClient {
+	restClient := rest.NewClient(
+		append([]rest.Option{
+			// Defaults to production stage if no option is supplied
+			WithStage(stages.StageProd),
+		}, opts...)...,
+	)
+
+	return &client{Client: restClient, clientID: clientID}
+}
+
+func (c *client) GetNode(ctx context.Context, id uuid.UUID) (models.GetNodeResponse, error) {
+	request := rest.Get("v2/nodes/{node}").
+		Assign("node", id).
+		SetHeader("Accept", "application/json").
+		SetHeader("X-Client-Id", c.clientID)
+
+	var response models.GetNodeResponse
+	if err := c.DoAndUnmarshal(ctx, request, &response); err != nil {
```

Comment:
How do we want to handle Problems with this client?

Reply:
I think we might need to do some changes to the main client library to support it. Doing it like this will remove all information regarding the Problem and just focus on the HTTP status code (400, 404, etc)